keanu-project/src/main/java/io/improbable/keanu/vertices/IVertex.java

```diff
@@ -0,0 +1,12 @@
+package io.improbable.keanu.vertices;
+
+import java.util.List;
+
+public interface IVertex<T, V extends IVertex<?, V>> {
```

Comment:
On reflection I'm not sure if this interface is useful; I could rip it out. The name is horrible too and was always meant to be temporary!

My intention was to separate the concept of Vertex (a thing with parents and a value) with the concept of a random variable (a thing that you can sample from etc.) but I sort of abandoned the idea halfway through the refactor, as I'm not sure its useful or desirable to separate the two concepts.